web/client/epics/geoProcessingTools.js

```diff
@@ -148,44 +146,64 @@ export const checkWPSAvailabilityGPTEpic = (action$, store) => action$
+                    return Rx.Observable.from([
+                        setWPSAvailability(layerId, true, source),
+                        getFeatures(layerId, source),
+                        checkingWPS(false)
+                    ]);
                 }
-                return Rx.Observable.from(actions);
+                // wps are all present, but
+                // we miss the describe feature type info so we trigger it and we wait for the answer
+
+                return Rx.Observable.from([
+                    setWPSAvailability(layerId, true, source),
+                    getDescribeLayer(layerUrl, layer, {}, source)
+                ])
+                    .merge(
+                        Rx.Observable.race(
+                            action$.ofType(UPDATE_NODE).filter(({node}) => node === layerId),
+                            Rx.Observable.timer(10 * 1000).mapTo("timeout")
+                        )
+                            .switchMap((act) => {
+                                // if the describe ft goes in timeout or has an error we invalidate the layer
+                                if (act === "timeout" || act?.error) {
+                                    return Rx.Observable.of(setInvalidLayer(layerId, source));
+                                }
+                                return Rx.Observable.of(
+                                    getFeatures(layerId, source),
+                                    checkingWPS(false));
+                            })
+                    );
             })
-            .catch(() => {
-                return Rx.Observable.of(setWPSAvailability(layerId, false, source), checkingWPS(false));
+            .catch((e) => {
+                console.error(e);
+                return Rx.Observable.of(
+                    setWPSAvailability(layerId, false, source),
+                    checkingWPS(false)
+                );
             })
             .startWith(checkingWPS(true));
     });
-/**
- * trigger the getFeatures stream given some parameters.
- * this needs to be done when a describe feature type is requested and the response is dispatched
- */
-export const triggerGetFeaturesGPTEpic = (action$) => action$
-    .ofType(DESCRIBE_FEATURE_TYPE_LOADED)
-    .switchMap(({layerId, source}) => {
-        return Rx.Observable.of(getFeatures(layerId, source));
-    });
-/**
- * handle the case when after the describe layer is executed we found out that
- * the selected layer is a raster.
- */
-export const disableCoverageLayerGPTEpic = (action$) => action$
-    .ofType(DESCRIBE_COVERAGES_LOADED)
-    .switchMap(({layerId}) => {
-        return Rx.Observable.of(setInvalidLayer(layerId, true));
-    });
 /**
  * fetch all features ids
  */
 export const getFeaturesGPTEpic = (action$, store) => action$
     .ofType(GET_FEATURES)
-    .switchMap(({layerId, source}) => {
+    .mergeMap(({layerId, source}) => {
```

Comment:
why changed to merge map?

Reply:
because when switching from buffer to intersection the sourcelayer and intersectionLayer were competing to trigget the epics **checkWPSAvailabilityGPTEpic** and **getFeaturesGPTEpic** doing this prevents the first one to stop and the update to the loading of source to layer to be stopped leaving it in a loading state forever.

Reply:
ok